packages/playwright-test/src/runner/taskRunner.ts

```diff
@@ -45,66 +45,70 @@ export class TaskRunner<Context> {
   }
 
   async run(context: Context, deadline: number): Promise<FullResult['status']> {
+    const { status, cleanup } = await this.runDeferCleanup(context, deadline);
+    await cleanup();
```

Comment:
It seems that we will lose the `status` from cleanup here? In the situation where setup+tests succeeded, but teardown failed/interrupted for whatever reason.